src/mlpack/methods/kde/kde_model_impl.hpp

```diff
@@ -214,6 +244,26 @@ inline void KDEModel::BuildModel(arma::mat&& referenceSet)
         (relError, absError, kernel::TriangularKernel(bandwidth));
   }
 
+  // Set whether to use Monte Carlo estimations or not.
+  MonteCarloVisitor MCVisitor(monteCarlo);
+  boost::apply_visitor(MCVisitor, kdeModel);
+
+  // Set Monte Carlo probability.
+  MCProbabilityVisitor probabilityVisitor(MCProb);
+  boost::apply_visitor(probabilityVisitor, kdeModel);
+
+  // Set Monte Carlo initial sample size.
+  MCSampleSizeVisitor sampleSizeVisitor(initialSampleSize);
+  boost::apply_visitor(sampleSizeVisitor, kdeModel);
+
+  // Set Monte Carlo entry coefficient.
+  MCEntryCoefVisitor entryCoefficientVisitor(MCEntryCoef);
+  boost::apply_visitor(entryCoefficientVisitor, kdeModel);
+
+  // Set Monte Carlo break coefficient.
+  MCBreakCoefVisitor breakCoefficientVisitor(MCBreakCoef);
+  boost::apply_visitor(breakCoefficientVisitor, kdeModel);
```

Comment:
Since all these are set together, you might consider writing just one visitor that sets all of the Monte Carlo parameters.  Up to you. :+1: